framework_tvb/tvb/core/services/project_service.py

```diff
@@ -582,22 +582,22 @@ def _remove_project_node_files(self, project_id, gid, skip_validation=False):
                                        datatype.parent_operation.range_values)
                     new_op = dao.store_entity(new_op)
                     to_project = self.find_project(links[0].fk_to_project)
-                    to_project_path = self.structure_helper.get_project_folder(to_project)
+                    to_project_path = self.storage_interface.get_project_folder(to_project.name)
 
-                    encryption_handler.set_project_active(to_project)
-                    encryption_handler.sync_folders(to_project_path)
+                    self.storage_interface.set_project_active(to_project)
```

Comment:
I think this could also be re-written so you would have a method in StorageInterface to take care of moving the links to the proper folders instead of these various calls to retrieve the project folder, sync for encryption, move files.